utils/report_generator.py

```diff
@@ -0,0 +1,197 @@
+        fw (flywheel.Client()): the flywheel client for the current session
+        acq_id (str): the flywheel ID of the parent acquisition of the target file
+        file_name (str): the name of the target file
+        DICOM_PRESENT (int): the number of DICOM images present in the target file
+        DICOM_SENT (int): the number of DICOM images successfully sent to the server
+        work_dir (pathlib.PosixPath): The directory to generate/look for an existing report in
+
+    """
+    
+    if not os.path.exists(work_dir):
+        os.makedirs(work_dir)
+    
+    report_file = Path(work_dir/'dicom-send-report.csv')
+    
+    if not os.path.exists(report_file):
+        log.info('Initializing report')
+        initialize_report(report_file)
+    
+    log.debug('Writing to report')
+    new_row = generate_list(fw, acq_id, file_name, DICOMS_PRESENT, DICOMS_SENT)
+    log.info(new_row)
+    append_list_as_row(report_file, new_row)
+    
+
+def print_report(report_file):
+    """Print the current report .csv file in a human-readable way to the log
+    
+    Args:
+        report_file (pathlib.PosixPath): the .csv file to print
+
+
+    """
+    
+    # This gets the lengths of each element in each row/col.  These lengths are then
+    # used to format the printed output so that it's human readable.
+    with open(report_file, 'r', newline='') as read_obj:
+        csv_f = reader(read_obj)
+        lens = [ [len(i) for i in row] for row in csv_f]
+    
+    max_lens = [max(idx) for idx in zip(*lens)]
+    # 4 spaces are added between columns (1/2 a standard tab)
+    format_string = ["{:<"+str(ml+4)+"}" for ml in max_lens]
+    format_string = " ".join(format_string)+"\n"
+        
+    with open(report_file, 'r', newline='') as read_obj:
+        csv_f = reader(read_obj)
+        print_string = 'REPORT_SUMMARY:\n'
+        for row in csv_f:
+            print_string+=format_string.format(*row)
+        
+        log.info(print_string)
+
+
+def upload_report(fw, ses_id, acq_id=None,
+                  report_file=Path('/flywheel/v0/report/dicom-send-report.csv')):
+    """Upload a target report .csv file to a session as an attachment
+    
+    Renames the report file based on the session and date.  If a single acquisition was specified
+    for export, the acquisition label is included too.
+    
+    Args:
+        fw (flywheel.Client()): the flywheel client for the current session
+        acq_id (str): the flywheel ID of the parent acquisition of the target file
+        ses_id (str): the flywheel ID of the parent session of the target file
+        report_file (str): the name of the target file
+
+    Returns:
+
+    """
+        
+    ses = fw.get_session(ses_id)
+    new_name = ses.label
+    
+    if acq_id is not None:
+        acq = fw.get_acquisition(acq_id)
+        new_name = f"{new_name}_{acq.label}"
+    
+    timestamp = datetime.now()
+    
+    new_name = f"{new_name}_{timestamp.strftime('%Y-%m-%d_%H:%M:%S')}.csv"
```

Comment:
Let's include the gear name here
```suggestion
    new_name = f"dicom-send_report-{new_name}_{timestamp.strftime('%Y-%m-%d_%H:%M:%S')}.csv"
```
I would be tempted to also include the job id, but I can't recall how readily accessible that is.